src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -495,5 +578,186 @@ private void LookupForMatchingGitHubRepository(BuildData mergedBuild)
+                if (splitIdx < 0)
+                {
+                    continue;
+                }
+
+                var key = item.Substring(0, splitIdx).Trim();
+                if (string.IsNullOrEmpty(key))
+                {
+                    continue;
+                }
+
+                var value = item.Substring(splitIdx + 1);
+                values[key] = value;
+            }
+
+            return values;
+        }
+
+        private void CreateAndPushMergedManifest(IImmutableList<AssetData> assets, SigningInformation finalSigningInfo)
+        {
+            string mergedManifestPath = Path.Combine(GetAzDevStagingDirectory(), MergeManifestFileName);
+
+            BuildModel buildModel = new BuildModel(
+                        new BuildIdentity
+                        {
+                            Attributes = GetNamedProperties(ManifestBuildData),
+                            Name = GetAzDevRepository(),
+                            BuildId = GetAzDevBuildNumber(),
+                            Branch = GetAzDevBranch(),
+                            Commit = GetAzDevCommit(),
+                            IsStable = IsStableBuild.ToString(),
+                            PublishingVersion = PublishingVersion
+                        });
+
+            foreach (AssetData data in assets)
+            {
+                if (blobSet.Contains(data.Name))
+                {
+                    BlobArtifactModel blobArtifactModel = new BlobArtifactModel
+                    {
+                        Attributes = new Dictionary<string, string>
+                                {
+                                    { "NonShipping", data.NonShipping.ToString().ToLower() }
+                                },
+                        Id = data.Name
+                    };
+                    buildModel.Artifacts.Blobs.Add(blobArtifactModel);
+                }
+                else
+                {
+                    PackageArtifactModel packageArtifactModel = new PackageArtifactModel
+                    {
+                        Attributes = new Dictionary<string, string>
+                                {
+                                    { "NonShipping", data.NonShipping.ToString().ToLower() }
+                                },
+                        Id = data.Name,
+                        Version = data.Version
+                    };
+                    buildModel.Artifacts.Packages.Add(packageArtifactModel);
+                }
+            }
+
+            XElement buildModelXml = buildModel.ToXml();
+
+            if (finalSigningInfo != null)
+            {
+                buildModelXml.Add(SigningInfoToXml(finalSigningInfo));
+            }
+
+            File.WriteAllText(mergedManifestPath, buildModelXml.ToString());
+
+            Log.LogMessage(MessageImportance.High,
+                        $"##vso[artifact.upload containerfolder=BlobArtifacts;artifactname=BlobArtifacts]{mergedManifestPath}");
```

Comment:
Should this value be extracted as a constant? Is the AssetManifests container more appropriate?

Reply:
This won't be reused so I don't want to declare a variable for a single use. Also, as previously discussed, this will end up in BlobArtifacts